# Analysing adult dataset

In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## The goal

The goal of this notebook is to analyse the adult dataset and to predict whether a person earns more than 50K a year or not.

In [15]:
df = pd.read_csv('adult.data', header=None) # header= None, otherwise the first row will be treated as column names

In [16]:
df.shape

(32561, 15)

Here we have a 32561 lines of data with 15 columns.

In [17]:
df.columns

Index([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14], dtype='int64')

In [18]:
df.columns = ['C' + str(i) for i in range(df.shape[1])] 

In [19]:
df.columns

Index(['C0', 'C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7', 'C8', 'C9', 'C10',
       'C11', 'C12', 'C13', 'C14'],
      dtype='object')

In [28]:
df.loc[:,['C14' , 'C0']]

C14  C0
0       <=50K  39
1       <=50K  50
2       <=50K  38
3       <=50K  53
4       <=50K  28
...       ...  ..
32556   <=50K  27
32557    >50K  40
32558   <=50K  58
32559   <=50K  22
32560    >50K  52

[32561 rows x 2 columns]

In [29]:
df.head()

C0                 C1      C2          C3  C4                   C5  \
0  39          State-gov   77516   Bachelors  13        Never-married   
1  50   Self-emp-not-inc   83311   Bachelors  13   Married-civ-spouse   
2  38            Private  215646     HS-grad   9             Divorced   
3  53            Private  234721        11th   7   Married-civ-spouse   
4  28            Private  338409   Bachelors  13   Married-civ-spouse   

                   C6              C7      C8       C9   C10  C11  C12  \
0        Adm-clerical   Not-in-family   White     Male  2174    0   40   
1     Exec-managerial         Husband   White     Male     0    0   13   
2   Handlers-cleaners   Not-in-family   White     Male     0    0   40   
3   Handlers-cleaners         Husband   Black     Male     0    0   40   
4      Prof-specialty            Wife   Black   Female     0    0   40   

              C13     C14  
0   United-States   <=50K  
1   United-States   <=50K  
2   United-States   <=50K  
3   United-States   <=50K  
4            Cuba   <=50K

In [30]:
LABEL_ANALYSE = df['C14'].unique()

In [32]:
LABEL_ANALYSE.shape

(2,)

The LABEL_ANALYSE has been created to analyse the label column. It has the unique information of a person does ou not a 50K a year.

In [33]:
idx = df['C14'] == LABEL_ANALYSE[0] # idx is a boolean array

In [34]:
df['C14'].loc[idx] = -1

/tmp/ipykernel_11442/3483701917.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['C14'].loc[idx] = -1


In [35]:
df['C14'].loc[~idx] = 1

/tmp/ipykernel_11442/1980075394.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['C14'].loc[~idx] = 1


In [36]:
df['C14']

0        -1
1        -1
2        -1
3        -1
4        -1
         ..
32556    -1
32557     1
32558    -1
32559    -1
32560     1
Name: C14, Length: 32561, dtype: object

In [37]:
df['C14'].unique()

array([-1, 1], dtype=object)

In [38]:
df.head()

C0                 C1      C2          C3  C4                   C5  \
0  39          State-gov   77516   Bachelors  13        Never-married   
1  50   Self-emp-not-inc   83311   Bachelors  13   Married-civ-spouse   
2  38            Private  215646     HS-grad   9             Divorced   
3  53            Private  234721        11th   7   Married-civ-spouse   
4  28            Private  338409   Bachelors  13   Married-civ-spouse   

                   C6              C7      C8       C9   C10  C11  C12  \
0        Adm-clerical   Not-in-family   White     Male  2174    0   40   
1     Exec-managerial         Husband   White     Male     0    0   13   
2   Handlers-cleaners   Not-in-family   White     Male     0    0   40   
3   Handlers-cleaners         Husband   Black     Male     0    0   40   
4      Prof-specialty            Wife   Black   Female     0    0   40   

              C13 C14  
0   United-States  -1  
1   United-States  -1  
2   United-States  -1  
3   United-States  -1  
4            Cuba  -1

In [41]:
df = pd.get_dummies(df, columns=['C1']) # get_dummies is a function in pandas to do one-hot encoding for categorical features. 
#It will create new columns for each unique value in the categorical feature.

df.shape

(32561, 23)

In [43]:
df = pd.get_dummies(df, columns=['C14'])

In [44]:
df.to_csv('adult.csv')